SE-XXX Typed throws.md

```diff
@@ -1,17 +1,22 @@
-Hello Swift Community. I started a conversation about this on the [Evolution|Proposals](https://forums.swift.org/t/typed-throw-functions/38860/) Forum and some people found interesting that might be worth to write a pitch about it.
+# Typed throws

```

Comment:
I am sticking to this proposal template now: https://github.com/apple/swift-evolution/blob/master/proposal-templates/0000-swift-template.md

Reply:
That's great to be found! Is very similar of what we've been doing but we can ultimately migrate all the content to that template!

Reply:
I'll take the parts that are missing from the structure and add it to the template we have now.